### ISSUE TITLE
Undefined name when using variable annotations

### ISSUE THREAD

**Issue by srittau** [PyCQA/pyflakes#486]
Please consider the following file:

```python
Type: object
def foo(x: Type) -> None: ...
```

When checking it with '2.1.1 Python 3.8.0 on Linux' (or Python 3.7.3) I get the following error:

```
foo.py:2: undefined name 'Type'
```

This is problematic when checking python stub files.

**Comment by asottile** [comment]
should that be:

```python
Type: object = ...
```

**Comment by srittau** [comment]
No, `Type: object` is correct syntax and actually preferred for type stubs, since the `= ...` part has no meaning.

**Comment by asottile** [comment]
`Type: object` doesn't introduce a name, which is why pyflakes (imo correctly) flags it as an undefined name later

**Comment by asottile** [comment]
```pycon
>>> Type: object
>>> def foo(x: Type) -> None: ...
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Type' is not defined
```

**Comment by srittau** [comment]
Good point, although pyflakes also fails when using `from __future__ import annotations`, which works with Python ≥ 3.7. Stubs are treated like this import was present.

**Comment by asottile** [comment]
how should `__future__` `annotations` be implemented? currently `pyflakes` has handling of it but it just defers the accesses until after the module has been parsed

**Comment by srittau** [comment]
@asottile Sorry, I don't think I understand the question. I have no experience with how pyflakes is implemented if that is what you mean? I'll investigate.

[some time later ...]

I only looked at it superficially for now, so I might be on the wrong track here. I see two potential areas that need to be changed: The `ANNASSIGN` handler currently ignores nodes without a value. That makes sense, since they don't generate a name. Nevertheless it should record the defined names somewhere (in parallel to `addBinding`) to be used in type annotations and potentially for #533 as well.

`handleNodeLoad` needs to be modified to use the new names record, but only in the context of type annotations. This is the hard part, since at that point it is not clear that we are parsing an annotation. Maybe a parse-level flag needs to be set while parsing annotations?

**Comment by asottile** [comment]
cool cool, there's an `_in_annotation` that should help with the second part

**Comment by CristinaBotez-Ipsos** [comment]
Hi! I'm using vim with the Ale plugin that's configured to use Flake8 in vim. I stumbled across an issue when searching for a way to configure Flake8 to either ignore annotations or consider the annotations imported from __future__.
What I'm trying to do is split a pretty big file into multiple modules without having to make a circular import or use forward annotations.

Here's an example:

Current file:
```python
from __future__ import annotations

class Mixin:
    def f(b: B):
        pass

class A(Mixin):
    pass

class B:
    pass
```

I'd like to split it into:
- The `mixins.py` file:
```python
from __future__ import annotations

class Mixin:
    def f(b: B):
        pass
```
- The `models.py` file:
```python
from __future__ import annotations
from .mixins import Mixin

class A(Mixin):
    pass

class B:
    pass
```
